src/realm/group.hpp

```diff
@@ -1002,7 +1002,7 @@ inline ref_type Group::get_history_ref(const Array& top) noexcept
 inline int Group::get_history_schema_version(const Array& top) noexcept
 {
     if (top.is_attached()) {
-        if (top.size() >= 8) {
+        if (top.size() >= 9) {
```

Comment:
This change is a consequence of allowing file format version 6 to be opened in "nonshared" mode (i.e., via Group rather than SharedGroup). I.e., it is a natural part of the change made by this PR.